### PR TITLE
Feat/create list and list-item components

### PR DESCRIPTION
**Pull request by 0xLodz**
## List and ListItem components

This PR  prepares the creation of the selected data list (#240), by introducing the generic `List` and `ListItems` components, designed to provide interactive and customisable list elements.
The components are built with the intention of being easily integrated into various parts of the application that display list items, such as dataset sharing.

## Usage
```tsx
const TestListExemple: FC = () => {
  const handleClick = () => {
    console.log('clicked')
  }
  return (
    <List>
      {[1, 2, 3].map((data, i) => (
        <ListItem key={i} leftElement={<div>left element</div>} onClick={handleClick} />
      ))}

      <ListItem />
      <ListItem
        content={<div>content!</div>}
        rightElement={
          <button className="okp4-dataverse-portal-button-main">
            X
          </button>
        }
      />
    </List>
  )
}
```

### CODE REVIEW

File: src/ui/component/list/list.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export const ListItem: FC<ListItemProps> = ({
+  leftElement,
+  rightElement,
+  content,
+  onClick,
+  className
+}) => (
+  <li
+    className={classNames('okp4-dataverse-portal-list-item-main', className, {
+      'pointer-cursor': onClick
```
Comment (fredericvilcot) [inline src/ui/component/list/list.tsx:22]: Maybe something like `clickable` would have been more appropriated here :)

File: src/ui/component/list/list.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+.okp4-dataverse-portal-list-item-main {
+  @include with-theme {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    padding: 12px 24px;
+    border-radius: 12px;
+    background-color: themed('list-item');
+    color: themed('text');
+    font-family: 'Noto Sans', sans-serif;
```
Comment (lolottetheclash) [inline src/ui/component/list/list.scss:19]: you should use our fonts mixins ;)

File: src/ui/component/list/list.tsx (hunk 1 above)
Comment (0xLodz) [reply src/ui/component/list/list.tsx]: Sounds a more appropriated name 👍

**Review by lolottetheclash — CHANGES_REQUESTED** [review]
It think you'd have to manage the space in a different way than via flex, as the elements are wrongly placed according to which you pass (see screenshot).
You'd also need to give them a bit of space between each other, because when the list item is shrunk, they're stuck together.
Capture d’écran 2023-06-27 à 09 19 06

**Review by ccamel — APPROVED** [review]
LGTM!

**Review by lolottetheclash — CHANGES_REQUESTED** [review]
I think that the problem persists if you look at the screenshots here:
Capture d’écran 2023-06-27 à 12 21 37

Also, when I'm looking at the figma, content should start right after the first element:

Capture d’écran 2023-06-27 à 12 20 49

**Comment by 0xLodz** [comment]
@lolottetheclash Thank you for your detailed feedback! The component has been updated according to the latest mockup. As we always have content, I made it required. It reduced the number of combinations of grid template columns from 9 to 3.

File: src/ui/component/list/list.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+export const ListItem: FC<ListItemProps> = ({
+  leftElement,
+  rightElement,
+  content,
+  onClick,
+  className
+}) => {
+  console.log('render')
```
Comment (fredericvilcot) [inline src/ui/component/list/list.tsx:30]: 🫣😝

Comment (0xLodz) [reply src/ui/component/list/list.tsx]: Oops sorry! Fixed.

**Review by fredericvilcot — APPROVED** [review]
LGTM! 👌

File: src/ui/component/list/list.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  content: ReactElement
+  leftElement?: ReactElement
+  rightElement?: ReactElement
+  onClick?: (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => void
+  className?: string
+}
+
+const gridTemplateColumnClasses = (
+  leftElement: ReactElement | undefined,
+  rightElement: ReactElement | undefined
```
Comment (lolottetheclash) [inline src/ui/component/list/list.tsx:15]: you can remove `undefined` and update it with optional 😉

**Review by lolottetheclash — CHANGES_REQUESTED** [review]
It's working well, last remarks and it will be good to go !

**Review by lolottetheclash — APPROVED** [review]
good for me, thanks !

**Pull request merged**